addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,766 @@
+
+        Returns a tuple with the pdf name, mimetype, content and field.
+        """
+        self.ensure_one()
+        file_data = self._l10n_vn_edi_fetch_invoice_file_data('PDF')
+        file_bytes = base64.b64decode(file_data['fileToBytes'])
+
+        return file_data['fileName'], 'application/pdf', file_bytes, 'l10n_vn_edi_sinvoice_pdf_file'
+
+    def action_l10n_vn_edi_update_payment_status(self):
+        """ Send a request to update the payment status of the invoice. """
+
+        # Lock
+        self._l10n_vn_edi_lock_records()
+
+        for invoice in self.filtered(lambda i: i.l10n_vn_edi_invoice_state == 'payment_state_to_update'):
+            sinvoice_status = 'unpaid'
+            endpoint = None
+            params = None
+
+            # SInvoice will return a NOT_FOUND_DATA error if the status in Odoo matches the one on their side.
+            # Because of that we wouldn't be able to differentiate a real issue (invoice on our side not matching theirs)
+            # With simply a status already up to date. So we need to check the status first to see if we need to update.
+            invoice_lookup, error_message = invoice._l10n_vn_edi_lookup_invoice()
+            if error_message:
+                raise UserError(error_message)
+
+            if 'result' in invoice_lookup:
+                invoice_data = invoice_lookup['result'][0]
+                if invoice_data['status'] == 'Chưa thanh toán':  # Vietnamese for 'unpaid'
+                    sinvoice_status = 'unpaid'
+                else:
+                    sinvoice_status = 'paid'
+
+            params = {
+                'supplierTaxCode': invoice.company_id.vat,
+                'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+            }
+
+            if invoice.payment_state in {'in_payment', 'paid'} and sinvoice_status == 'unpaid':
+                # Mark the invoice as paid
+                endpoint = 'InvoiceAPI/InvoiceWS/updatePaymentStatus'
+                params['templateCode'] = invoice.l10n_vn_edi_invoice_symbol.invoice_template_id.name
+            elif invoice.payment_state not in {'in_payment', 'paid'} and sinvoice_status == 'paid':
+                # Mark the invoice as not paid
+                endpoint = 'InvoiceAPI/InvoiceWS/cancelPaymentStatus'
+            else:
+                continue
+
+            _request_response, error_message = invoice._l10n_vn_edi_send_request(
+                method='POST',
+                endpoint=endpoint,
+                params=params,
+            )
+
+            if error_message:
+                raise UserError(error_message)
+
+            # Revert back to the sent state as the status is up-to-date.
+            invoice.l10n_vn_edi_invoice_state = 'sent'
+
+    def _need_cancel_request(self):
+        # EXTEND 'account'
+        return super()._need_cancel_request() or self._l10n_vn_edi_is_sent()
```

Comment:
For example here, it's equivalent to
```
        return super()._need_cancel_request() or self.l10n_bn_edi_invoice_state in ('sent', 'payment_state_to_update', 'canceled', 'adjusted', 'replaced')
```

Knowing the need_cancel_request is called when the invoice is posted, I deduce you could have a posted invoice being 'sent', 'payment_state_to_update', 'canceled', 'adjusted' or 'replaced'.

Reply:
Which makes sense to me :thinking: 
It'll be called when the state changes, which is when you'd expect the buttons to be update.
We don't want users to reset to draft invoices that are still active on the platform, the case that could make sense is if the invoice was cancelled. (And even there, I would personally just keep the cancelled invoice and make a new one)